MainApp/lib/app/routes/api.js

```diff
@@ -29,9 +29,12 @@ export default function (router) {
         trialinfo.username = req.body.username;
         trialinfo.trialid = req.body.trialid;
         trialinfo.condition = req.body.condition;
-        
-        if (trialinfo.username == null || trialinfo.username == '' || trialinfo.trialid == null || trialinfo.trialid == '' || trialinfo.condition == null ||trialinfo.condition == '') {
-            res.send({success: false, message: 'Username or trialid or condition was empty' + trialinfo.username + ' ID:' +trialinfo.trialid + 'CON:' + trialinfo.condition});
+
+        if (trialinfo.username == null || trialinfo.username == '' || trialinfo.trialid == null || trialinfo.trialid == '' || trialinfo.condition == null || trialinfo.condition == '') {
```

Comment:
mongoose `required: true` works as expected. I'm waiting for Mehrdad's reply for validation so I'm going to be changing the schema soon. I'll probably end up removing these lines if it's easier to do via mongoose. These cases were there in the tutorial which is why all of us have used them.